### PR TITLE
Use modal dialog for OS project selection

### LINKED ISSUE #208: Use tooltip / dropdown for a client project's list of open source projects
Instead of showing them in the table, create a tooltip or dropdown that shows them all.

cursor_and_bitcentive

### PR DESCRIPTION
**Pull request by DesignByOnyx**
Closes #208 

I went through several iterations and had to settle on the modal.

- Semantic UI didn't work. I tried both Marshall's [semantic-ui-dropdown-canjs](https://github.com/icanjs/semantic-ui-dropdown-canjs) and vanilla Semantic UI - to no avail on either. Didn't want to spend cycles on _why_.
- Bootstrap Tooltips only work on :hover
- [Bootstrap Popovers](http://getbootstrap.com/javascript/#live-demo-1) aren't good with interactive content as there's no "click outside" to close popovers without doing [stuff like this](http://stackoverflow.com/questions/11703093/how-to-dismiss-a-twitter-bootstrap-popover-by-clicking-outside). You have to click on the button to both open **and close** the popover, which was bad UX even for me knowing this limitation.  

os-project-select

### CODE REVIEW

File: public/components/client-projects/client-projects.stache [hunk 1]
```diff
@@ -21,30 +21,62 @@
...
-                       {{#if commissioned}} class="commissioned" {{/if}}>
-                  {{osProjectRef.value.name}}
-                </label>
-            {{/each}}
+        <button type="button" class="btn btn-link btn-sm" data-toggle="modal" data-target="#modal_{{id}}">
+          Edit OS Projects
+        </button>
+
+        {{! Modal }}
+        <div class="modal fade" id="modal_{{id}}" tabindex="-1" role="dialog" aria-hidden="true">
```
Comment (justinbmeyer) [inline public/components/client-projects/client-projects.stache]: move into partial

Comment (justinbmeyer) [inline public/components/client-projects/client-projects.stache]: `_id`?

File: public/components/client-projects/client-projects.stache [hunk 2]
```diff
@@ -21,30 +21,62 @@
...
+                  </li>
+                  {{/each}}
+                </ul>
+              </div>
+            </div>
           </div>
-        {{/if}}
+        </div>
+
+        {{! List of selected OS Projects}}
```
Comment (justinbmeyer) [inline public/components/client-projects/client-projects.stache]: this should go away.

File: public/components/client-projects/client-projects-os-projects-modal.stache [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        </button>
+        <h4 class="modal-title">Monthy Client OS Projects</h4>
+      </div>
+      <div class="modal-body">
+        <ul class="os-project-list">
+        {{#each contributionMonth.monthlyOSProjects as monthlyOSProject}}
+          <li>
+            <input id="os_project_{{osProjectRef._id}}"
+                   type="checkbox"
+                   {{#if monthlyClientProjectsOSProjects.has(monthlyOSProject) }}checked{{/if}}
```
Comment (justinbmeyer) [inline public/components/client-projects/client-projects-os-projects-modal.stache]: `{$checked}="monthlyClientProjectsOSProjects.has(monthlyOSProject)"`

File: public/components/client-projects/client-projects-os-projects-modal.stache [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                   value="{{osProjectRef.value.name}}"
```
Comment (justinbmeyer) [inline public/components/client-projects/client-projects-os-projects-modal.stache]: {$value}="osProjectRef.value.name"

File: public/components/client-projects/client-projects-os-projects-modal.stache [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                   ($click)="toggleUseProject(monthlyClientProjectsOSProjects, monthlyOSProject)"
+                   {{#if contributionMonth.isSaving}} disabled{{/if}}/>
```
Comment (justinbmeyer) [inline public/components/client-projects/client-projects-os-projects-modal.stache]: {$disabled}

File: public/components/client-projects/client-projects-os-projects-modal.stache [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<div class="modal fade" id="os-project-modal" tabindex="-1" role="dialog" aria-hidden="true">
+  <div class="modal-dialog valign-center" role="document">
+    {{#with selectedClientProject}}
```
Comment (justinbmeyer) [inline public/components/client-projects/client-projects-os-projects-modal.stache]: Can you remove `with`.

File: public/components/client-projects/client-projects.stache [hunk 7]
```diff
@@ -58,10 +44,14 @@
...
     </tr>
+    {{/if}}
   {{/each}}
-  {{/with}}
   </tbody>
 </table>
+{{/with}}
+
+{{! Partial defined in the view-model }}
+{{>os-projects-modal}}
```
Comment (justinbmeyer) [inline public/components/client-projects/client-projects.stache]: Do this via `<can-import>`?

```js
<can-import from="./os-projects-modal.stache" {^@value}="@*osProjectsModal"/>
{{>*osProjectsModal}}
```

```js
<can-import from="./os-projects-modal.stache" {^value}="*osProjectsModal"/>
{{>*osProjectsModal}}
```

File: public/components/client-projects/client-projects.js [hunk 8]
```diff
@@ -17,6 +21,9 @@ export const ClientProjectVM = DefineMap.extend({
       return ClientProject.getList({});
     }
   },
+  selectedClientProject: {
```
Comment (justinbmeyer) [inline public/components/client-projects/client-projects.js]: `selectedClientProject: ClientProject`

**Comment by DesignByOnyx** [comment]
os-project-select

File: public/components/client-projects/client-projects.html [hunk 9]
```diff
@@ -1,4 +1,6 @@
 <script type="text/stache" can-autorender id="demo">
+  <can-import from="jquery" />
```
Comment (justinbmeyer) [inline public/components/client-projects/client-projects.html]: why is jquery needed?

**Pull request merged**